packages/mattermost-components/package.json

```diff
@@ -0,0 +1,58 @@
+{
+  "name": "@mattermost/components",
+  "module": "dist/index.esm.js",
+  "types": "dist/index.esm.d.ts",
+  "scripts": {
+    "build": "rollup -c",
+    "debug": "rollup -c"
+  },
+  "workspaces": [
+    "../mattermost-redux",
+    "../reselect",
+    "../types"
+  ],
+  "devDependencies": {
+    "@babel/cli": "^7.17.6",
+    "@babel/core": "^7.17.7",
+    "@babel/plugin-transform-runtime": "^7.17.0",
+    "@babel/preset-env": "^7.16.11",
+    "@babel/preset-react": "^7.16.7",
+    "@babel/preset-typescript": "^7.16.7",
+    "@rollup/plugin-babel": "^5.3.1",
+    "@rollup/plugin-commonjs": "^21.0.2",
+    "@rollup/plugin-node-resolve": "^13.1.3",
+    "@rollup/plugin-typescript": "^8.3.1",
+    "@types/lodash": "^4.14.178",
+    "@types/react": "16.9.35",
+    "@types/react-dom": "16.9.8",
+    "@types/react-redux": "7.1.21",
+    "@types/shallow-equals": "^1.0.0",
+    "@types/styled-components": "5.1.19",
+    "babel-loader": "^8.2.3",
+    "babel-plugin-formatjs": "10.3.14",
+    "babel-plugin-styled-components": "^2.0.6",
+    "css-loader": "^6.7.1",
+    "rollup": "^2.70.1",
+    "rollup-plugin-auto-external": "^2.0.0",
+    "rollup-plugin-peer-deps-external": "^2.2.4",
+    "rollup-plugin-scss": "^3.0.0",
+    "rollup-plugin-ts": "^2.0.5",
+    "sass": "^1.49.9",
+    "sass-loader": "^12.6.0",
+    "style-loader": "^3.3.1",
+    "typescript": "4.3.4",
```

Comment:
We should probably be less particular with some of these since some will conflict with the root package.json. The rest of these are probably fine since they have the `^`, but this and some of the peer dependencies might cause issues when we go to update stuff.

Separate from this, I'm also realizing we might be adding more work to our regular dependency updates with packages like this. Hopefully it's not too bad though since these should be fairly similar dependencies to what we use everywhere else

Reply:
Not sure what exactly we should do about this dependency management. For module federation to work we want these to stay in sync.

Reply:
I think it's easier to keep them in sync though if we aren't strict about the exact versions needed so that the web app can be an authority on the versions of everything. If we specify exact versions, then I think that means we're more likely to load multiple versions of a library

Reply:
With the no lock file in the package this works.